rocks-lib/src/build/mod.rs

```diff
@@ -0,0 +1,70 @@
+use crate::{
+    config::Config,
+    rockspec::{Build, BuildBackendSpec, RockSourceSpec, Rockspec},
+    tree::Tree,
+};
+use eyre::Result;
+use git2::Repository;
+
+pub fn build(rockspec: Rockspec, config: &Config, no_install: bool) -> Result<()> {
```

Comment:
suggestion(readability, non-blocking): I'm not really a fan of "negative" boolean parameters.

```suggestion
pub fn build(rockspec: Rockspec, config: &Config, install: bool) -> Result<()> {
```

Perhaps it would be even nicer to omit the parameter and extract the logic into a separate function.

Reply:
Neither am I, I was following the function signature that `luarocks` uses here. I thought it could come in handy once we start working on the compatibility scripts for various "installer" rocks (like `luarocks-build-rust-mlua`). The function invoked there does supply a `no_install`.

I can only assume it was created that way because under 90% of cases you want to build and install, it's very unlikely you'll want to just build. I'm happy to remove this in general though in order make the code cleaner.

Reply:
What I did for now is I removed the `no_install` flag from the library API (it's completely useless there), but kept it in the internal build code for the time being. Then, depending on how the compatibility layer pans out, we can remove or alter that flag.